test/azure_blobs_exception_tests.jl

```diff
@@ -144,3 +144,141 @@
+
+        try
+            method === :GET && blob_get!(joinpath(baseurl, "blob"), zeros(UInt8, 5), creds)
+            method === :PUT && blob_put(joinpath(baseurl, "blob"), codeunits("a,b,c"), creds)
+            @test false # Should have thrown an error
+        catch e
+            @test e isa ErrorException
+            @test occursin(string(response_status), e.msg)
+            response_status < 500 && (@test occursin("response body from the dummy server", e.msg))
+        finally
+            close(http_server)
+        end
+        wait(http_server)
+        return nretries[]
+    end
+
+    # See https://learn.microsoft.com/en-us/rest/api/searchservice/http-status-codes
+
+    @testset "400: Bad Request" begin
+        # Returned when there's an error in the request URI, headers, or body. The response body
+        # contains an error message explaining what the specific problem is.
+        nretries = test_status(:GET, 400)
+        @test nretries == 1 broken=true
+        nretries = test_status(:PUT, 400)
+        @test nretries == 1 broken=true
+    end
+
+    @testset "403: Forbidden" begin
+        # Returned when you pass an invalid api-key.
+        nretries = test_status(:GET, 403)
+        @test nretries == 1 broken=true
+        nretries = test_status(:PUT, 403)
+        @test nretries == 1 broken=true
+    end
+
+    @testset "404: Not Found" begin
+        nretries = test_status(:GET, 404)
+        @test nretries == 1
+    end
+
+    @testset "405: Method Not Supported" begin
+        nretries = test_status(:GET, 405, ["Allow" => "PUT"])
+        @test nretries == 1 broken=true
+        nretries = test_status(:PUT, 405, ["Allow" => "GET"])
+        @test nretries == 1 broken=true
+    end
+
+    @testset "409: Conflict" begin
+        # Returned when write operations conflict.
+        # NOTE: We currently don't retry but maybe we should? This is probably a case where the
+        # retry logic should add more noise to the backoff so that multiple writers don't collide on retry.
+        nretries = test_status(:GET, 409)
+        @test nretries == max_retries broken=true
+        nretries = test_status(:PUT, 409)
+        @test nretries == max_retries broken=true
+    end
+
+    @testset "412: Precondition Failed" begin
+        # Returned when an If-Match or If-None-Match header's condition evaluates to false
+        nretries = test_status(:GET, 412)
+        @test nretries == 1
+        nretries = test_status(:PUT, 412)
+        @test nretries == 1
+    end
+
+    @testset "413: Content Too Large" begin
+        # https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob?tabs=shared-access-signatures#remarks
+        nretries = test_status(:PUT, 413)
+        @test nretries == 1 broken=true
+    end
+
+    @testset "429: Too Many Requests" begin
+        # TODO: We probably should respect the Retry-After header, but we currently don't
```

Comment:
Is this something that HTTP.jl handles now? This might be a high priority thing for us to implement.

Reply:
Okay, i think this is the only one to decide on... currently we're not retrying 429 at all, but arguably we should be (and if so, should be respecting the `Retry-After` header if set)

Reply:
i'll leave this to a follow-up PR, if/when we have more info about if we need to support this at all and if it's best done inthe ObjectStore.jl repo (rather than the application itself)